pkg/compact/compact_test.go

```diff
@@ -174,3 +178,360 @@ func BenchmarkGatherNoCompactionMarkFilter_Filter(b *testing.B) {
 	}
 
 }
+
+type planResult struct {
```

Comment:
Can we make this struct as an inner struct in the `testPlannerRanges` test case?